### PR TITLE
remove more lingering bazel references

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

More cleanup following https://github.com/kubernetes/enhancements/issues/2420

Now that < 1.20 are long out of support we can do more cleanup. Noticed one of these in https://github.com/kubernetes/kubernetes/pull/112819#discussion_r986353374

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -1991,10 +1991,6 @@ rules:
       dir: staging/src/k8s.io/pod-security-admission
   library: true
 recursive-delete-patterns:
-- BUILD
-- '*/BUILD'
-- BUILD.bazel
-- '*/BUILD.bazel'
```
Comment (BenTheElder) [inline staging/publishing/rules.yaml:1997]: /assign @nikhita

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -1991,10 +1991,6 @@ rules:
...
 - Gopkg.toml
```
Comment (BenTheElder) [inline staging/publishing/rules.yaml:1994]: @nikhita: can we delete this now as well? I can't imagine we've had a Gopkg.toml for ... ages now 🤔

File: .generated_files [hunk 3]
```diff
@@ -16,7 +16,6 @@
 
 file-prefix	zz_generated.
 
-file-name	BUILD
```
Comment (BenTheElder) [inline .generated_files:19]: /assign @thockin

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (dims) [reply staging/publishing/rules.yaml:1994]: hehe yep

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (BenTheElder) [reply staging/publishing/rules.yaml:1994]: dep shouldn't be used by anyone anymore as well filed https://github.com/kubernetes/kubernetes/pull/112870

**Pull request merged**